src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApClientTPC.java

```diff
@@ -61,6 +61,42 @@ public class MeasurementBasedApClientTPC extends TPC {
 	/** The target MCS index. */
 	private final int targetMcs;
 
+	/** Factory method to parse generic args map into the proper constructor */
+	public static MeasurementBasedApClientTPC makeWithArgs(
+		DataModel model,
+		String zone,
+		DeviceDataManager deviceDataManager,
+		Map<String, String> args
+	) {
+		int targetMcs = DEFAULT_TARGET_MCS;
+
+		String arg;
+		if ((arg = args.get("targetMcs")) != null) {
+
+			try {
+				targetMcs = Integer.parseInt(arg);
+			} catch (NumberFormatException e) {
+				logger.error(
+					"Invalid integer passed to parameter targetMcs, using default value: ",
+					e
+				);
+			}
+			if (targetMcs < 0) {
+				logger.error(
+					"Invalid value passed for targetMcs - must be greater than 0. Using default value."
+				);
+				targetMcs = DEFAULT_TARGET_MCS;
+			}
+		}
```

Comment:
nit: to avoid needing to reassign the default (which looks messy), I would set a temp variable first and do range checks on that. can also move this into the try-catch block if you want (to limit the temp var scope).